### PR TITLE
add arbitrum/optimism chain tvl to defisaver

### PR DESCRIPTION
**Pull request by kfukue**
Hello,
I picked up this ticket (https://github.com/DefiLlama/DefiLlama-Adapters/projects/2#card-84919189) to add the tvl and borrowed amount from defisaver in optimism and arbitrum. I read the events of the DefisaverLogger contract to see which proxy contracts were used to interact with Defisaver to calculate the tvl and borrowed amount that were used in AaveV3 (using AaveV3View contract's getLoanDataArr method). 
I used the collateral amount as tvl and the borrowed amounts (both borrowedStableAmounts and borrowVariableAmounts) as borrowed. I also did not make any changes to the calculation of the tvl for mainnet(ethereum).

Please review this when you have the chance and let me know if you have any questions or feedback.
Thanks!

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/defisaver.js exports TVL: 
        
 
 
    ethereum                  131.51 M
    optimism-borrowed         9.00 M
    optimism                  13.78 M
    arbitrum-borrowed         460.01 k
    arbitrum                  1.22 M
    borrowed                  9.46 M
    
    total                    146.51 M

File: projects/defisaver.js [hunk 1]
```diff
@@ -325,8 +363,156 @@ async function tvl(ts, block) {
...
+    const startBlock = networkMapping[chainId].startBlock;
+    const provider = getProvider(chain);
+    const endBlock = await provider.getBlockNumber();
+    const topic = "RecipeEvent(address,string)";
+    const topic1 = "ActionDirectEvent(address,string,bytes)";
+    const loggerContract = getAddressByChain(DefisaverLogger, chainId);
+    const aaveV3ViewAddress = getAddressByChain(AaveV3View, chainId);
+    // get proxy contracts by looking through the events : RecipeEvent(address,string)
+    const recipeEvents = (
+      await sdk.api.util.getLogs({
```
Comment (g1nt0ki) [inline projects/defisaver.js:404]: @kfukue thanks for the PR, fetching logs is bit of an issue on any chain that is not Ethereum. These queries become slower over time and stop working eventually. So, I reject any PRs with this, sorry.

Can you check if they have some sort of graph api for fetching this information?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]

File: projects/defisaver.js (hunk 1 above)
Comment (kfukue) [reply projects/defisaver.js:404]: @g1nt0ki No worries. Yea that makes sense. Ok, I'll reach out to the defisaver team to see if there's a graph api or an alternative way to get these proxy contract addresses but if there isn't I'll close the PR until we have a better way to get the proxy contract addresses. Thanks for your feedback!

Comment (kfukue) [reply projects/defisaver.js:404]: @g1nt0ki I'll close this for now. (not much response from their team)

Comment (g1nt0ki) [reply projects/defisaver.js:404]: @kfukue yeah, sometimes it is quite frustrating.

**Pull request closed without merging**